domain/consensus/processes/blockbuilder/block_builder.go

```diff
@@ -89,6 +97,53 @@ func (bb *blockBuilder) buildBlock(coinbaseData *externalapi.DomainCoinbaseData,
 	}, nil
 }
 
+func (bb *blockBuilder) validateTransactions(transactions []*externalapi.DomainTransaction) error {
+	invalidTransactions := make([]ruleerrors.InvalidTransaction, 0)
+	for _, tx := range transactions {
+		err := bb.validateTransaction(tx)
+		if err != nil {
+			if !errors.As(err, &ruleerrors.RuleError{}) {
+				return err
+			}
+			invalidTransactions = append(invalidTransactions,
+				ruleerrors.InvalidTransaction{Transaction: tx, Error: err})
+		}
+	}
+
+	if len(invalidTransactions) > 0 {
+		return ruleerrors.NewErrInvalidTransactionsInNewBlock(invalidTransactions)
+	}
+
+	return nil
+}
+
+func (bb *blockBuilder) validateTransaction(tx *externalapi.DomainTransaction) error {
+	originalEntries := make([]externalapi.UTXOEntry, len(tx.Inputs))
+	for i, input := range tx.Inputs {
+		originalEntries[i] = input.UTXOEntry
+		input.UTXOEntry = nil
+	}
+
+	defer func() {
+		for i, input := range tx.Inputs {
+			input.UTXOEntry = originalEntries[i]
+		}
+	}()
```

Comment:
What's the point of re-populating the transactions with the original UTXOEntries?